packages/Discretization/test/MeshGenerator.hpp

```diff
@@ -0,0 +1,493 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_MESHGENERATOR_HPP
+#define DTK_VIEWTESTCUDAHELPERS_HPP
+
+#include <DTK_CellTypes.h>
+
+#include <Kokkos_Core.hpp>
+#include <Teuchos_DefaultComm.hpp>
+
+template <typename DeviceType>
+Kokkos::View<double **, DeviceType>
+computeCoordinates( unsigned int const n_vertices,
+                    std::vector<unsigned int> const &n_subdivisions,
+                    unsigned int comm_rank )
+{
+    unsigned int const dim = n_subdivisions.size();
+
+    Kokkos::View<double **, DeviceType> coordinates( "coordinates", n_vertices,
+                                                     dim );
+    auto coordinates_host = Kokkos::create_mirror_view( coordinates );
+    std::vector<unsigned int> current_vertex( dim, 0 );
+    unsigned int proc_offset = n_subdivisions[dim - 1] * comm_rank;
+    for ( unsigned int i = 0; i < n_vertices; ++i )
+    {
+        for ( unsigned int d = 0; d < dim; ++d )
+        {
+            unsigned int pos = current_vertex[d];
+            unsigned int j = ( d == dim - 1 ) ? ( proc_offset + pos ) : pos;
```

Comment:
:@ Please document the function and explain that you are offsetting in the last dimension and that points are regularly spaced with grid spacing 1.